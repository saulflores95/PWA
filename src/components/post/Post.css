.title {
  font-size:3rem;
  color: gray;
}

.content {

}
