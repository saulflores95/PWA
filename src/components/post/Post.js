import React from 'react'
import styles from './Post.css'

const Post = () => (
  <div>
    <h2 className={styles.title}><a className={styles.link} href='#'>Blog Title</a></h2>
    <p className={styles.content}>Content Content Content Content Content Content Content Content Content Content Content Content Content </p>
  </div>
)

export default Post
