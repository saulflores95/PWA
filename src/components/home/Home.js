import React from 'react'
import Post from '../post/Post'

const Home = ({posts}) => (
  <div>
    {posts.map(post => <Post key={post.slug} {...post} titleLink />)}
  </div>
)

export default Home
