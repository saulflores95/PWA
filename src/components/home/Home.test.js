import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Home from './Home'
import Post from '../Post/Post'


test('Home component should render a single post', () => {
  const posts = {
    posts: [{
      title: 'How to build a progressive web app with React',
      slug: 'how-to-build-a-progressive-web-app-with-react',
      excerpt: 'One of the easiest ways to get a job',
      content: 'One of the easiest ways to get a job in the early GitHub days...'
    }]
  };
  const component = shallow(<Home posts={posts.posts} />)
  const tree = toJson(component)
  expect(component.find(Post).length).toBe(1)
  expect(tree).toMatchSnapshot()
})

test('Home component should render two posts', () => {
  const posts = {
    posts: [{
      title: 'How to build a progressive web app with React',
      slug: 'how-to-build-a-progressive-web-app-with-react',
      excerpt: 'One of the easiest ways to get a job',
      content: 'One of the easiest ways to get a job in the early GitHub days...'
    }, {
      title: 'How to build a progressive web app with React',
      slug: 'how-to-build-a-progressive-web-app-with-react',
      excerpt: 'One of the easiest ways to get a job',
      content: 'One of the easiest ways to get a job in the early GitHub days...'
    }]
  };
  const component = shallow(<Home posts={posts.posts} />)
  const tree = toJson(component)
  expect(component.find(Post).length).toBe(2)
  expect(tree).toMatchSnapshot()
})

test('Home component should render "No posts here" if no posts were supplied', () => {
  const component = shallow(<Home posts={[]} />)
  const tree = toJson(component)

  expect(component.find(Post).length).toBe(0)
  expect(component.contains('No posts here')).toBe(true)
  expect(tree).toMatchSnapshot()
})
