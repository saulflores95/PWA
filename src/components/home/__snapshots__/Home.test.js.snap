exports[`test Home component should render "No posts here" if no posts were supplied 1`] = `
<div>
  <p>
    No posts here
  </p>
</div>
`;

exports[`test Home component should render a single post 1`] = `
<div>
  <Post
    content="One of the easiest ways to get a job in the early GitHub days..."
    excerpt="One of the easiest ways to get a job"
    slug="how-to-build-a-progressive-web-app-with-react"
    title="How to build a progressive web app with React"
    titleLink={true} />
</div>
`;

exports[`test Home component should render two posts 1`] = `
<div>
  <Post
    content="One of the easiest ways to get a job in the early GitHub days..."
    excerpt="One of the easiest ways to get a job"
    slug="how-to-build-a-progressive-web-app-with-react"
    title="How to build a progressive web app with React"
    titleLink={true} />
  <Post
    content="One of the easiest ways to get a job in the early GitHub days..."
    excerpt="One of the easiest ways to get a job"
    slug="how-to-build-a-progressive-web-app-with-react"
    title="How to build a progressive web app with React"
    titleLink={true} />
</div>
`;
